server/test/unit/server/webservices/views/test_consumers.py

```diff
@@ -0,0 +1,1377 @@
+                         'bindings': []}]
+        mock_resp.assert_called_once_with(expected_cont)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch('pulp.server.webservices.views.consumers.generate_redirect_response')
+    @mock.patch(
+        'pulp.server.webservices.views.consumers.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.consumers.factory.consumer_manager')
+    def test_create_consumer(self, mock_factory, mock_resp, mock_redirect):
+        """
+        Test consumer creation.
+        """
+        cons = {'id': 'foo', 'display_name': 'bar'}
+        cert = '12345'
+        expected_cont = {'consumer': {'id': 'foo', 'display_name': 'bar',
+                         '_href': '/v2/consumers/foo/'}, 'certificate': '12345'}
+
+        request = mock.MagicMock()
+        request.body = json.dumps({'id': 'foo', 'display_name': 'bar'})
+        mock_factory.return_value.register.return_value = cons, cert
+        consumers = ConsumersView()
+        response = consumers.post(request)
+        mock_resp.assert_called_once_with(expected_cont)
+        mock_redirect.assert_called_once_with(mock_resp.return_value,
+                                              expected_cont['consumer']['_href'])
+        self.assertTrue(response is mock_redirect.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    def test_create_consumer_missing_param(self):
+        """
+        Test consumer creation with missing required id.
+        """
+        request = mock.MagicMock()
+        request.body = json.dumps({'display_name': 'bar'})
+        consumers = ConsumersView()
+        try:
+            response = consumers.post(request)
+        except MissingValue, response:
+            pass
+        else:
+            raise AssertionError("MissingValue should be raised with missing options")
+        self.assertEqual(response.http_status_code, 400)
+        self.assertEqual(response.error_data['property_names'], ['id'])
+
+
+class TestConsumerResourceView(unittest.TestCase):
+    """
+    Test consumer resource view.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.consumers.generate_json_response')
+    @mock.patch('pulp.server.webservices.views.consumers.factory')
+    def test_delete_consumer_resource(self, mock_factory, mock_resp):
+        """
+        Test consumer delete resource.
+        """
+        mock_consumer_manager = mock.MagicMock()
+        mock_factory.consumer_manager.return_value = mock_consumer_manager
+        mock_consumer_manager.unregister.return_value = None
```

Comment:
If the structure changes so that "None" is returned rather than the return value of the manager, this line will not be necessary.